controlplane/kubeadm/controllers/upgrade.go

```diff
@@ -105,7 +106,10 @@ func (r *KubeadmControlPlaneReconciler) upgradeControlPlane(
 		return ctrl.Result{}, err
 	}
 
-	if status.Nodes <= *kcp.Spec.Replicas {
+	// RolloutStrategy is currently defaulted and validated to be RollingUpdate
```

Comment:
Could we add a check in code, maybe a switch case and make it fail if it's not RollingUpdate?

Reply:
Do we really need to have that? Can't we rely on defaulting and checks in webhook, and assume that RolloutStrategy can never be anything other than RollingUpdate? Any thoughts @detiber

Reply:
If we ever add more strategies, the code would be stronger to support those longer term

Reply:
Good point!